pkg/cloudevents/transport/pubsub/codec.go

```diff
@@ -187,16 +187,16 @@ func (c Codec) fromAttributes(a map[string]string, event *cloudevents.Event) err
 	delete(a, prefix+"source")
 
 	if t := a[prefix+"time"]; t != "" {
```

Comment:
This looks suspiciously similar to the amqp codec code. Perhaps there needs to be helpers in the main code for the codecs to use.

Reply:
they use a slightly different mapping ... grrrr

I have some ideas how to combine them and pass a mapping decoder ring for the future.